tests/testthat/test-class.R

```diff
@@ -60,6 +60,26 @@ describe("inheritance", {
   })
 })
 
+describe("abstract classes", {
+  it("can't be instantiated", {
+    expect_snapshot(error = TRUE, {
+      foo <- new_class("foo", abstract = TRUE)
+      foo()
+    })
+  })
+  it("can't inherit from concrete class", {
+    expect_snapshot(error = TRUE, {
+      foo1 <- new_class("foo1")
+      new_class("foo2", parent = foo1, abstract = TRUE)
+    })
+  })
+  it("can constructor concrete subclasses", {
+    foo1 <- new_class("foo1", abstract = TRUE)
+    foo2 <- new_class("foo2", parent = foo1)
+    expect_s3_class(foo2(), "foo2")
+  })
+})
+
```

Comment:
I think I was also expecting a few tests showing that you can create methods for abstract classes, and then have concrete subclasses that dispatch off those methods

Reply:
I didn't test that because I think it's highly unlikely that it wouldn't work given the current (and any future) implementation.